Sming/Components/OtaUpgrade/component.mk

```diff
@@ -0,0 +1,192 @@
+# Future configurability (ATM, there is only signing)
+
+# If enabled, OTA firmware images are digitally signed and the upgrade mechanism will reject
+# images with an invalid signature.
+# You should not disable this feature unless there is absolutely no way your device is accessed
+# through a public network.
+COMPONENT_VARS += OTA_ENABLE_SIGNING
+OTA_ENABLE_SIGNING ?= 1
+
+# If enabled, OTA firmware images are encrypted.
+# It is strongly encouraged to enable this feature if your firmware images contain secret information
+# (WiFi credentials, server certificates, etc.)
+# However, keep in mind the inherent weekness of this approach: An attacker with physical access to your device
+# can always extract the secret information, INCLUDING the private decryption key, which breaks all encryptions
+# using the same key (i.e. future firmware images of this project, other projects using the same key...)
+# Due to this weakness, encryption is disabled by default. You should think twice if this meachnism provides
+# the necessary security for your application or if another method (e.g. TLS) is more appropriate.
+COMPONENT_VARS += OTA_ENABLE_ENCRYPTION
+OTA_ENABLE_ENCRYPTION ?= 0
+
+# If enabled, the build timestamps of the current firmware and the OTA blob are not compared, 
+# thus allowing to downgrade to an older firmware version.
+# You are strongly advised to keep this option disabled, because otherwise a security fix could 
+# simply be reverted by downgrading to an unpatched firmware version.
+COMPONENT_VARS += OTA_ENABLE_DOWNGRADE
+OTA_ENABLE_DOWNGRADE ?= 0
+
+COMPONENT_INCDIRS := .
+
+COMPONENT_DOXYGEN_INPUT := .
+
+OTATOOL := python $(COMPONENT_PATH)/otatool.py
+
+ifneq ($(OTA_ENABLE_SIGNING),0)
+OTA_CRYPTO_FEATURES += --signed
+# has to be global, because it is used in a public header file
+GLOBAL_CFLAGS += -DOTA_SIGNED
```

Comment:
Should be `ifeq ($(OTA_ENABLE_SIGNING),1)` according to convention and docs.
Also, use same name for -D

Reply:
I should have said `ENABLE_OTA_SIGNING`. Confusing myself now...

Reply:
I deliberately chose conditions to do least harm if unexpected values are used for the definitions. E.g. if the user accidentally uses `OTA_ENABLE_SIGNING=ON` the original version of the condition would keep signing enabled, which (in my opinion) is more important from a security perspective than not disabling it for `OTA_ENABLE_SIGNING=OFF`.

Will change it anyway, just wanted to explain my thoughts.

Reply:
In this instance perhaps restrict the value to a choice of `1` or `0` and throw a tantrum at anything else?